src/Util/FileHasher.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Util;
+
+use Rector\Core\Exception\ShouldNotHappenException;
+
+final class FileHasher
+{
+    /**
+     * cryptographic insecure hasing of a string
+     */
+    public function hash(string $string): string
+    {
+        return hash($this->getAlgo(), $string);
+    }
+
+    /**
+     * cryptographic insecure hasing of files
+     *
+     * @param string[] $files
+     */
+    public function hashFiles(array $files): string
+    {
+        $configHash = '';
+        $algo = $this->getAlgo();
+        foreach ($files as $file) {
+            $hash = hash_file($algo, $file);
+            if ($hash === false) {
+                throw new ShouldNotHappenException(sprintf('File %s is not readable', $file));
+            }
+
+            $configHash .= $hash;
+        }
+
+        return $configHash;
+    }
+
+    private function getAlgo(): string
+    {
+        //see https://php.watch/articles/php-hash-benchmark
+        if (\PHP_VERSION_ID >= 80100) {
+            // if xxh128 is available use it, as it is way faster
+            return 'xxh128';
+        }
+
+        return 'crc32b';
```

Comment:
I prefer md4 based on discussion on original PR it created

Reply:
I don't agree but reverted to md4 ;)